app/controllers/summaries_controller.rb

```diff
@@ -3,5 +3,14 @@ def index
     @categories = Category.all
     @date_start = Purchase.minimum(:date)
     @date_end = Date.today
+    @totals = category_totals
+  end
+
+  def category_totals
+    @totals = []
+    Category.all.each do |c|
+      @totals << [c.name, Purchase.where(subcategory: c.subcategories).to_a.map(&:amount).sum]
+    end
+    @totals
```

Comment:
`@totals` shouldn't be an instance variable - it doesn't need to be, and you're assigning the return value of this method to `@totals` above.

Also, could this be accomplished simply with a map function? Haven't tested it, but something like
```ruby
Category.all.map do |cat|
   [ cat.name, Purchase.where(subcategory: cat.subcategories).to_a.map(&:amount).sum ]
end
```

That way you'd end up with the necessary array of arrays.

Reply:
Also, I actually wonder if it'd be useful to factor out `Purchase.where(subcategory: c.subcategories).to_a.map(&:amount).sum` into a separate method e.g. `sum_by_category`, simply for clarity.